frontend/localebuilder/check-extracted-messages.ts

```diff
@@ -0,0 +1,38 @@
+import fs from "fs";
+import path from "path";
+import childProcess from "child_process";
+import PO from "pofile";
+import { isDeepEqual } from "../lib/util/util";
+import chalk from "chalk";
+
+function getPoMessages(poText: string) {
+  const po = PO.parse(poText);
+  const obj: { [key: string]: string } = {};
+  for (let item of po.items) {
+    obj[item.msgid] = item.msgstr.join("");
+  }
+  return obj;
+}
+
+export function checkExtractedMessagesSync(poPath: string, extractCmd: string) {
+  const readPoSync = () =>
+    getPoMessages(fs.readFileSync(poPath, { encoding: "utf-8" }));
+  const relPath = path.relative(process.cwd(), poPath);
+  console.log(`Reading ${relPath}.`);
+  const origPo = readPoSync();
+  console.log(`Running "${extractCmd}"...`);
+  childProcess.execSync(extractCmd);
+  console.log(`Reading ${relPath} again.`);
+  const newestPo = readPoSync();
+  if (!isDeepEqual(origPo, newestPo)) {
+    console.log(
+      chalk.redBright(
+        `Extracted message catalogs are out of date! ` +
+          `Please re-run "${extractCmd}", or commit the changes to .po ` +
+          `files that have just been made on this machine.`
+      )
```

Comment:
Nice! At first glance of this, it wasn't fully clear to me that this command `yarn lingui:check` was also fully running the `extract` command as well and updating our messages.po files, perhaps because my assumption is that any "check" command is essentially "returning a boolean" in that it doesn't change anything, just determines whether something or not something is as we want it to be. That being said, it makes sense that of course you'd need to run the `extract` command in order to do the check. Also, the fact that the results get saved is very helpful as my default next step from seeing a failed check would be to run `extract` again and commit the updates. 

So, all in all, my only potential suggestion would be to either rename the command to be more explicit about also running the extract command in full (something like "extract-and-check"? Although that sounds a bit wordy), and/or changing the wording of the warning message on failed checks to be clearer that a full extract took place (maybe something like "We extracted the message catalogs and they are out of date! Please commit these changes or ...")?

Hmm, now I'm feeling like this is super knit-picky and maybe a non-monday-morning brain would've interpreted things better. Feel free to ignore these suggestions if the current implementation fits other similar features!

Reply:
No I think your criticisms totally make sense!  It was a bit frustrating to implement actually because there's no way (as far as I can tell) to figure out whether everything is copacetic _without_ actually running extract--my original intent was to make "check" just return a boolean but not make any modifications to the filesystem, so your initial assumption matches my intent, but then I found out there's not actually a way to do it without having "side effects" so to speak.

I like the idea of `yarn:extract-and-check` though! It is a bit verbose but that's OK because it's only really run in CircleCI, so I'll go ahead and make that change.